camel/functions/search_functions.py

```diff
@@ -0,0 +1,104 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import List
+
+import requests
+from bs4 import BeautifulSoup
+
+from .openai_function import OpenAIFunction
+
+
+def clean_str(p: str) -> str:
+    r"""Cleans the input string by encoding and decoding it multiple times
+    to ensure it can be properly read and processed by Python code
+
+    Args:
+        p (str): The input string to be cleaned, typically from the webpage
+
+    Returns:
+        str: The cleaned string
+    """
+    return p.encode().decode("unicode-escape")\
+            .encode("latin1").decode("utf-8")
+
+
+def get_page_obs(page):
+    r"""Returns the first 5 pages in the fetched page.
+
+    Returns:
+        List[str]: The list of the first 5 sentences
+    """
+    paragraphs = page.split('\n')
+    paragraphs = [p.strip() for p in paragraphs if p.strip()]
+
+    # find all sentences
+    sentences = []
+    for p in paragraphs:
+        sentences += p.split('. ')
+    sentences = [s.strip() + '.' for s in sentences if s.strip()]
+
+    return ' '.join(sentences[:5])
```

Comment:
Sorry this is not the number of available pages but the first 5 sentences in this page. I think it would be better to add another parameter to control the number of sentences to be returned.

Reply:
Well I thought hardcoding 5 sentences may be good... If I add another parameter to control the number of sentences, the model will try to adjust this and may not have a good control? Even if there are less than 5 pages (though such a Wiki page seems not existing), the range indexing can still work to return all sentences.

Reply:
Sure, no prob